pkgs/games/rotp/default.nix

```
@@ -0,0 +1,53 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, jdk, jre }:
+
+stdenv.mkDerivation {
+  pname = "remnants-of-the-precursors";
+  version = "1.04";
+
+  src = fetchFromGitHub {
+    owner = "rayfowler";
+    repo = "rotp-public";
+    rev = "e3726fc22c2c44316306c50c79779e3da1c4c140";
+    sha256 = "sha256-oMA8LRpBoBX7t4G+HuRz0a8g+UEwYO7Ya0Qq4j8AWec=";
+  };
+
+  nativeBuildInputs = [ jdk makeWrapper ];
+
+  # By default, the game tries to write to the java class path. If that fails
+  # (and it always does, since they are in the read-only nix store), it won't
+  # launch.
+  patches = [ ./0001-store-config-and-saves-in-XDG_CONFIG_HOME.patch ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    javac -d "$out/share/" -sourcepath src src/rotp/Rotp.java
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # We need the assets, but don't want the source files.
+    cp -r src/rotp/* "$out/share/rotp/"
+    find "$out/share/rotp" -type f -name '*java' -exec rm "{}" \;
+
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/rotp \
+      --add-flags "-cp $out/share rotp.Rotp"
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = ''Open-source modernization of the 1993 classic "Master of Orion", written in Java'';
+    homepage = "https://www.remnantsoftheprecursors.com/";
+    license = {
+      fullName = "ROTP License";
+      url = "https://github.com/rayfowler/rotp-public/blob/e3726fc22c2c44316306c50c79779e3da1c4c140/LICENSE";
```

```suggestion
      url = "https://github.com/rayfowler/rotp-public/blob/master/LICENSE";
```
Isn't this just by-nc-nd 4.0, asl 2.0, gpl3?

---

I used a specific hash instead of pointing to master in case the license changes. Seemed more robust.

---

> Isn't this just by-nc-nd 4.0, asl 2.0, gpl3?

Yeah, I think you're right. The only one that isn't obvious to me is the "Planet Generator" licence... and unfortunately the provided link doesn't work.

Let me know what the preferred course of action is here.

---

We should list all the licenses we know and can add a custom one for panet generator.

Also I think we can use src.rev to make sure we point to the right thing.

---

Ok, I got really specific with the licenses.